### PR TITLE
Rule: Useless Cast and not null assertion

### LINKED ISSUE #65: Rule: Useless Cast and not null assertion
The TypeScript compiler automatically casts objects to a specific type inside conditionals with instanceof :
```
if(x instanceof string) {
  x = (x as string).concat("postfix"); // This cast is not required
}
```
but also when property values that are set at type definition are checked for :

```
class Wolf extends Animal {
  kind = hunting.PackHunter;
  howl() {}
}

class Cat extends Animal {
  kind = hunting.SoloHunter;
  meow() {}
}

function emitSound(animal: Animal) {
  if(animal.kind === hunting.PackHunter) {
    (animal as Wolf).howl(); // This case is not required either
  }
}
```

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #65

### CODE REVIEW

File: sonarts-core/docs/rules/no-useless-cast.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# no-useless-cast
+
+The TypeScript compiler automatically casts a variable to the relevant type inside conditionals where it is possible to infer (usage of `typeof`, `instanceof` etc). These compiler feature makes casts and not-null assertions unnecessary.
```
Comment (saberduck) [inline sonarts-core/docs/rules/no-useless-cast.md]: @vilchik-elena  use singular "this compiler feature"

File: sonarts-core/src/utils/parser.ts [hunk 2]
```diff
@@ -35,8 +35,7 @@ export function parseString(source: string, scriptKind: ts.ScriptKind = ts.Scrip
  * @throws if parsing error
  */
 export function parseFile(filename: string): { sourceFile: ts.SourceFile; program: ts.Program } {
-  const compilerOptions = ts.getDefaultCompilerOptions();
-  compilerOptions.target = TARGET;
+  const compilerOptions = { strict: true, target: TARGET };
```
Comment (saberduck) [inline sonarts-core/src/utils/parser.ts]: do we need both this as `tsconfig.json` ?

File: sonarts-core/docs/rules/no-useless-cast.md (hunk 1 above)
Comment (vilchik-elena) [reply sonarts-core/docs/rules/no-useless-cast.md]: fixed

File: sonarts-core/src/utils/parser.ts (hunk 2 above)
Comment (vilchik-elena) [reply sonarts-core/src/utils/parser.ts]: removed

**Pull request merged**